### ISSUE TITLE
Confidence indicator

### ISSUE THREAD

**Issue by nareddyt** [nareddyt/discover-rewards-notifier#75]
Currently, we have 2 different ways of matching sites to discover deals or cashback rewards:

- **URL Matching:** (High Confidence) We check if any deals or cashbacks urls match the url of the website
- **Fuzzy Site Name Matching:** (Low Confidence) We do a fuzzy check on the name of the website. We try to find a deal or cashback that _contains_ all of the site title (with some irrelevant edge cases, see #70 and #67).

When we find a match and display a popup, we should display a confidence level in the popup UI. Specifically, we would want to add a UI component to the `templates/deal.handlebars` and `templates/cashback.handlebars` to display the confidence level of the deal or cashback.

I imagine this indicator would just list the confidence level and provide an information icon. When the user hovers over the information icon, it can show a [Bootstrap Popover](https://getbootstrap.com/docs/4.1/components/popovers/) with details on why it's high / low confidence. This is open to change though, feel free to post UI mockups below!

It would be nice if this is coded to support other confidence levels in the future.

**Comment by jyk88** [comment]
hi there. i'd like to work on this, but i have a question before i get started...

i did the following so far:
1. forked the repo and cloned it down to my local machine
2. followed npm install etc on README.md and then got it as a chrome extension
3. when i go to www.jcrew.com or www.ae.com as you suggest in README.md though, i don't get anything "lighting up" on the chrome extension icon. When I hover over the icon, it says "no discover deal for this page."

my question is -- is there a specific site (maybe jcrew.com is outdated?) or do i need to be a discover card member / log in and then i can see the deals?

thank you! sorry i am new to this, so i appreciate your help in advance ^_

**Comment by jyk88** [comment]
Ah! Okay, so here's what Inspection Views -> Background Page shows in the console...

> background.js:48 Cashback data has never been loaded! Extension will not work
> onCashbackData @ background.js:48

was i supposed to load cashback data somewhere in the install?

**Comment by nareddyt** [comment]
Nope, those should be loaded automatically when the extension runs. Currently, it loads those directly from [this url](https://tejunareddy.com/discover-rewards-notifier/cashbacks.json). Is it possible the network you are on is blocking requests to that URL when you visit it from your web browser? If so, let me know and I can think of a solution. That's a problem I haven't considered before!

My other guess is that GitHub pages (the site for hosting the data) was down 2 days ago. The status is [reported here](https://status.github.com/messages). That would have causes the extension to fail loading the data. However, the extension is designed to keep trying to load the data every hour. So even if it failed 2 days ago, it should have kept trying until it worked.

You can also try **reloading** the extension. On the Chrome Extensions page (three dots button -> more tools -> extensions), try hitting the refresh button under this extension. This should cause the extension to restart. Then when you open up the logs, they should be pretty clean (they should only have a few items). If there are are warnings or errors, can you paste them here?

**Comment by jyk88** [comment]
ah, i guess you did not see my edit on my last comment.

now i am seeing this instead of the console error in the console

> Fetched and loaded the latest Discover data successfully :)
> background.js:108 Registered all listeners :)

yay! thanks :) i'll try reloading the extension if i run into issues in the future.

**Comment by nareddyt** [comment]
Oh interesting, yeah I missed that. I wonder why it happened, but I guess it's hard to reproduce. But yay, it works :)

**Comment by spikerheado1234** [comment]
Hey, is this issue still open? If so i'd like to work on it.

**Comment by jyk88** [comment]
Hi Ahan,

Yeah go for it. Sorry I’ve been away for long, I won’t be able to work on
this so feel free to take it.
